PetrolEngine.cpp

```diff
@@ -12,9 +17,12 @@ PetrolEngine::PetrolEngine(int power, float capacity, int gears)
 
 void PetrolEngine::changeGear(int gear)
 {
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
+    if(gear < REAR_GEAR || gear > gears_){
+        throw GearException("This car don't have " + std::to_string(gear) + " gear.\n");
+    }
+    if(abs(gear - currentGear_) > 2) {
```

Comment:
I suggest changing magic '2' to const variable. It would be easier to change program if changing gears rules would be changed.